lib/srv/ctx.go

```diff
@@ -449,28 +450,18 @@ func (c *ServerContext) AddCloser(closer io.Closer) {
 
 // GetAgent returns a agent.Agent which represents the capabilities of an SSH agent.
 func (c *ServerContext) GetAgent() agent.Agent {
-	c.RLock()
-	defer c.RUnlock()
-	return c.agent
+	if c.Parent == nil {
```

Comment:
Should `GetAgent` and `GetAgentChannel` return an error? Because it could potentially segfault here.

Reply:
The fact that `agent` might be `nil` seems to be well handled, so I've opted to update the method docs to indicate that it might be `nil` rather than changing the method to return an error.